subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ikg_signature.c

```diff
@@ -158,28 +161,34 @@ int cracen_ikg_sign_digest(int identity_key_index, const struct sxhashalg *hasha
 
 int cracen_ikg_create_pub_key(int identity_key_index, uint8_t *pub_key)
 {
-	int status;
+	int status = SX_ERR_RETRY;
 	struct sx_pk_acq_req pkreq;
+	while (status != SX_OK) {
+		pkreq = sx_pk_acquire_req(SX_PK_CMD_IK_PUBKEY_GEN);
+		if (pkreq.status) {
+			return pkreq.status;
+		}
+		pkreq.status = sx_pk_list_ik_inslots(pkreq.req, identity_key_index, NULL);
+		if (pkreq.status) {
+			return pkreq.status;
+		}
 
-	pkreq = sx_pk_acquire_req(SX_PK_CMD_IK_PUBKEY_GEN);
-	if (pkreq.status) {
-		return pkreq.status;
-	}
-	pkreq.status = sx_pk_list_ik_inslots(pkreq.req, identity_key_index, NULL);
-	if (pkreq.status) {
-		return pkreq.status;
-	}
-
-	sx_pk_run(pkreq.req);
-	status = sx_pk_wait(pkreq.req);
-	if (status != SX_OK) {
-		return status;
+		sx_pk_run(pkreq.req);
+		status = sx_pk_wait(pkreq.req);
+		if (status == SX_ERR_RETRY) {
+			status = exit_ikg(&pkreq);
```

Comment:
Do we need to do that for the non cracen lite devices? Do you know if it has any unwanted side effects?

Also this while loop could be infinite in case of  continues error, does it make sense to have a limit for repetitions here as well?

Reply:
We don't, but the alternative is either 1. creating a separate function for cracen lite, or 2. ifdefing it out. Both of those make the code messier which is why I just added the while loop. I tried running the code on l15 and it didn't affect the runtime, but I haven't checked power usage. The only side effect I can see would be if SX_ERR_RETRY was returned from CRACEN, in which case it would loop instead immediately returning, but I don't see that as necessarily being worse?

I will add a limit on repetitions for the key generation

Reply:
I agree that the code will be messier for sure, I am just trying to figure out if there is any chance that we will end up with unwanted side effects. But since you will add the limit on the repetitions it should be easy to find out so that should be fine.